internal/pkg/ccloudv2/flink_gateway.go

```diff
@@ -106,10 +107,58 @@ func (c *FlinkGatewayClient) GetStatementResults(environmentId, statementId, org
 		req = req.PageToken(pageToken)
 	}
 	resp, httpResp, err := req.Execute()
-	return resp, errors.CatchCCloudV2Error(err, httpResp)
+	return resp, makeFlinkError(err, httpResp)
 }
 
 func (c *FlinkGatewayClient) GetExceptions(environmentId, statementId, orgId string) (flinkgatewayv1alpha1.SqlV1alpha1StatementExceptionList, error) {
 	resp, httpResp, err := c.StatementExceptionsSqlV1alpha1Api.GetSqlV1alpha1StatementExceptions(c.flinkGatewayApiContext(), environmentId, statementId).OrgId(orgId).Execute()
-	return resp, errors.CatchCCloudV2Error(err, httpResp)
+	return resp, makeFlinkError(err, httpResp)
+}
+
+// FlinkError extends the ErrorWithSuggestion with a status code.
+type FlinkError struct {
+	errorMsg       string
+	suggestionsMsg string
+	statusCode     int
+}
+
+func NewFlinkError(errorMsg string, suggestionsMsg string, statusCode int) FlinkError {
+	return FlinkError{
+		errorMsg:       errorMsg,
+		suggestionsMsg: suggestionsMsg,
+		statusCode:     statusCode,
+	}
+}
+
+func (f FlinkError) StatusCode() int {
+	return f.statusCode
+}
+
+func (f FlinkError) GetSuggestionsMsg() string {
+	return f.suggestionsMsg
+}
+
+func (f FlinkError) Error() string {
+	return f.errorMsg
+}
+
+type Coder interface {
+	StatusCode() int
+}
+
+var _ Coder = (*FlinkError)(nil)
+var _ errors.ErrorWithSuggestions = (*FlinkError)(nil)
+
+// Extends error with status code, including suggestion if err type is ErrorWithSuggestion
+func makeFlinkError(err error, r *http.Response) error {
+	err = errors.CatchCCloudV2Error(err, r)
```

Comment:
I think you need to early return here in case `errors.CatchCCloudV2Error(err, r)` returns nil, otherwise you can get a NPE in line 161

Reply:
ops, fixing